### PR TITLE
Add ShellEx support for Windows & fix issues (#20216)

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Port of #20216 for issue #18949

* Add ShellEx support for Windows & fix issues

Adds ShellExecute support for Windows (outside of UAP). Unlike desktop
UseShellExecute is not the default.

In the process of ding this I ran across two issues:

1- The owned process handle would be disposed after hitting members on
Process. Getting the ProcessName, for example, would cause a closed
handle exception on the next call that used the handle.

2- The MainWindowTitle import wasn't set to the right charset. In addition
the allocated buffer size wasn't correct. Fixed and added debug checks
for failed api calls.

* Address feedback

- Add usings in tests
- Handle missing API

* Move to new using pattern

- Remove newly dead string
- Other feedback

* More feedback changes

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Plan is for @tarekgh to submit to shiproom tomorrow pending daily run results. I'll be OOF until 5/31.

**Comment by tarekgh** [comment]
@JeremyKuhne I'll take care of it. please send me the info you want to send to the ship room.

File: src/System.Net.Http/tests/FunctionalTests/Configurations.props [hunk 1]
```diff
@@ -6,6 +6,7 @@
       netcoreapp-Unix;
       netstandard-Windows_NT;
       netstandard-Unix;
+      uap-Windows_NT;
```
Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: Why did you include this Http stuff? Shouldn't that be a different PR?

**Comment by JeremyKuhne** [comment]
Adding minor test change that I have a dependency on. https://github.com/dotnet/corefx/commit/8455655dc0ee18d361d1669cb45fb191189c82f6

cc: @davidsh

File: src/System.Net.Http/tests/FunctionalTests/Configurations.props (hunk 1 above)
Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: Oh I see.

Comment (JeremyKuhne) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: See my other comment- I'm dependent on that change for the PlatformDetection. This change is benign from a 2.0 perspective.

Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: I don't think it is necessary to include these two files though, only PlatformDetection. These guys can be reverted.

Comment (JeremyKuhne) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: Not strictly necessary, but taking the change as is seems cleaner and more traceable. Given that the change is trivial that seems the better choice to me.

Comment (JeremyKuhne) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: Assuming, of course, you Rebase and Merge rather than Squash and Merge.

Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/Configurations.props]: fine

**Comment by tarekgh** [comment]
#20289
